index.js

```diff
@@ -2,10 +2,23 @@
 
 // Lambda Insight Layer Versions
 // see https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Lambda-Insights-extension-versions.html
-const latestLayerVersion = 14;
-const layerVersions = [2, 11, 12, 14];
-const layerArn = (region, version) =>
-  `arn:aws:lambda:${region}:580247275435:layer:LambdaInsightsExtension:${version}`;
+const layerVersions = require('./layerVersions.json');
+
+const layerArn = (region, version) => {
+  if (version) {
+    // TODO: this does not appropriately handle non 'aws' partitions
```

Comment:
we can use something like 

```
try {
        const layerVersionInfo = await this.provider.request('Lambda', 'getLayerVersionByArn', {
          Arn: 'arn:aws:lambda:${this.region}:580247275435:layer:LambdaInsightsExtension:${version}',
        });
        return layerVersionInfo.LayerVersionArn;
      } catch (err) {
        throw new Error(
            'LambdaInsights layer version '${version}'' +
            'does not exist within your region '${this.region}'.');
      }
```

But I would suggest to create another PR for this implementation.